material-select/src/dotMenu.ets

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2022 Application Library Engineering Group.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { MenuProperties } from './properties'
+import { DotMenuOption } from './menuOptions'
+
+@Component
+struct DotMenu {
+	menu: DotMenuOption[] = []
+	onSelect: (it: DotMenuOption) => void
+	@State model: DotMenu.Model = new DotMenu.Model()
+	@State menuModel: DotMenu.MenuModel = new DotMenu.MenuModel()
+
+	aboutToAppear() {
+		if (this.model == null || this.model == undefined) {
+			this.model = new DotMenu.Model()
+		}
+		if (this.menuModel == null || this.menuModel == undefined) {
+			this.menuModel = new DotMenu.MenuModel()
+		}
+		if (this.menu == null || this.menu == undefined) {
+			this.menu = []
+		}
+	}
+
+	@Builder
+	MenuItem(option: DotMenuOption) {
+		Row() {
+			Image(option.getIcon())
+				.objectFit(ImageFit.Contain)
+				.width(this.menuModel.getIconWidth())
+				.height(this.menuModel.getIconHeight())
+				.margin({right: 12})
+			Text(option.getValue())
+				.fontSize(this.menuModel.getFontSize())
+				.fontColor(this.menuModel.getFontColor())
+				.flexGrow(1)
+				.maxLines(2)
+				.textOverflow({ overflow: TextOverflow.Ellipsis })
+				.fontWeight(FontWeight.Bold)
+		}
+		.height(this.menuModel.getHeight())
+		.width(this.menuModel.getWidth())
+		.onClick(() => {
+			if (this.onSelect != null && this.onSelect != undefined) this.onSelect(option)
+		})
+		.padding(this.menuModel.getPadding())
+	}
+
+	@Builder
+	MenuBuilder() {
+		Column() {
+			List() {
+				ForEach(this.menu, (item) => {
+					ListItem() {
+						this.MenuItem(item)
+					}
+				}, item => item.id)
+			}
+		}
+		.width(this.menuModel.getWidth())
+		.padding({ top: 7, bottom: 7 })
+		.borderRadius(this.menuModel.getBorderRadius())
+	}
+
+	build() {
+		Image($r('app.media.dot_menu'))
```

Comment:
all the images / resources used in the library needs to be placed in the resource folder within te library, pls move them from entry to library